Sources/PackageModel/PackageReference.swift

```diff
@@ -27,61 +28,64 @@ public struct PackageReference: Codable {
         case remote
     }
 
-    /// Compute the default name of a package given its URL.
-    public static func computeDefaultName(fromURL url: String) -> String {
-      #if os(Windows)
-        let isSeparator : (Character) -> Bool = { $0 == "/" || $0 == "\\" }
-      #else
-        let isSeparator : (Character) -> Bool = { $0 == "/" }
-      #endif
-
-        // Get the last path component of the URL.
-        // Drop the last character in case it's a trailing slash.
-        var endIndex = url.endIndex
-        if let lastCharacter = url.last, isSeparator(lastCharacter) {
-            endIndex = url.index(before: endIndex)
-        }
-
-        let separatorIndex = url[..<endIndex].lastIndex(where: isSeparator)
-        let startIndex = separatorIndex.map { url.index(after: $0) } ?? url.startIndex
-        var lastComponent = url[startIndex..<endIndex]
-
-        // Strip `.git` suffix if present.
-        if lastComponent.hasSuffix(".git") {
-            lastComponent = lastComponent.dropLast(4)
-        }
-
-        return String(lastComponent)
-    }
-
     /// The identity of the package.
     public let identity: PackageIdentity
 
-    /// The name of the package, if available.
-    public var name: String {
-        _name ?? Self.computeDefaultName(fromURL: path)
-    }
-    private let _name: String?
-
     /// The path of the package.
     ///
     /// This could be a remote repository, local repository or local package.
-    public let path: String
+    public let location: String
 
     /// The kind of package: root, local, or remote.
     public let kind: Kind
 
+    // FIXME
+    /// An alternate identity of the package.
+    /// This would be deprecated when identity refactoring is complete.
+    /// Right now, there is a way to "override" the identity of
+    /// the package from the name in the manifest, this is a crutch until we remove
+    /// the name from the manifest all together
```

Comment:
You're referring to the `name` parameter of a dependency here, right?

Reply:
> You're referring to the name parameter of a dependency here, right?

there are two things we need to change address as we make progress on the identity front:
1. explicit identity for dependencies (today we have the optional `name` parameter but we need something "stronger" to replace it
2. get rid of the `name` attribute in the manifest which we today lean on to drive identity but is give its free form nature can very easily lead to identity conflicts

Reply:
I think it might be good to make this more explicit in the comment, the casual reader might think of `Package.name` here and get confused.